chainerchem/dataset/preprocessors/weavenet_preprocessor.py

```diff
@@ -1,17 +1,166 @@
+import os
+
 import numpy
 from rdkit import Chem
+from rdkit import RDConfig
+from rdkit.Chem import AllChem
+from rdkit.Chem import ChemicalFeatures
 
 from chainerchem.dataset.preprocessors.common \
     import construct_atomic_number_array
 from chainerchem.dataset.preprocessors.common import type_check_num_atoms
 from chainerchem.dataset.preprocessors.mol_preprocessor import MolPreprocessor
 
 
-MAX_DISTANCE = 7
-# TODO(Nakago): Review default size
-DEFAULT_NUM_MAX_ATOMS = 20  # 40
+ATOM = ['H', 'C', 'N', 'O', 'S', 'Cl', 'Br', 'F', 'P', 'I']
+MAX_DISTANCE = 2  # 7
+DEFAULT_NUM_MAX_ATOMS = 20  # 60  # paper
+
+
+# --- Atom feature extraction ---
+def construct_atom_type_vector(mol, num_max_atoms=DEFAULT_NUM_MAX_ATOMS):
+    n_atom_type = len(ATOM)
+    n_atom = mol.GetNumAtoms()
+    atom_type_vector = numpy.zeros((num_max_atoms, n_atom_type),
+                                   dtype=numpy.float32)
+    for i in range(n_atom):
+        a = mol.GetAtomWithIdx(i)
+        atom_idx = ATOM.index(a.GetSymbol())
```

Comment:
1. sorry i haven't tested yet.
can we use `try ... except raise MolExtractError` for this type of error handling?

2. Noted. I will update naming convention based on your suggestion.

3. Yes, I'm thinking that this is general feature extraction function and it might be ok that user or other preprocessor will use it in the future.

Reply:
> can we use try ... except raise MolExtractError for this type of error handling?

Yes. Then, please add tests for this function that checks this behavior.